transport/src/main/java/io/camunda/zeebe/transport/stream/impl/RemoteStreamImpl.java

```diff
@@ -48,41 +44,82 @@ public M metadata() {
 
   @Override
   public void push(final P payload) {
-    executor.execute(
-        () -> {
-          final List<StreamConsumer<M>> streamConsumers = stream.streamConsumers();
-          final var randomClient =
-              streamConsumers.get(ThreadLocalRandom.current().nextInt(streamConsumers.size()));
-          streamer.pushAsync(
-              payload,
-              (error, p) -> {
-                final var consumers = new ArrayList<>(streamConsumers);
-                consumers.remove(randomClient);
-                Collections.shuffle(consumers);
-                new RetryHandler(errorHandler, consumers).retry(error, payload);
-              },
-              randomClient.id());
-        });
+    final var initialConsumer = pickInitialConsumer();
+    if (initialConsumer == null) {
+      errorHandler.handleError(
```

Comment:
:thought_balloon: This will be run in the engine thread. But I guess this is ok, because we know the errorHandler is non-blocking eventhough we don't enforce it.

Reply:
:thinking: Should we enforce it? We could use the streamer's executor if only to avoid additional load on the engine. Seems a little much considering it then delegates to another actor, but :shrug: It's a pretty uncommon case, I would lean to leave it as is.